pkg/vault/contracts/authorizer/TimelockAuthorizer.sol

```diff
@@ -830,7 +885,12 @@ contract TimelockAuthorizer is IAuthorizer, IAuthentication, ReentrancyGuard {
         for (uint256 i = 0; i < executors.length; i++) {
             // Note that we allow for repeated executors - this is not an issue
             _isExecutor[scheduledExecutionId][executors[i]] = true;
-            emit ExecutorCreated(scheduledExecutionId, executors[i]);
+            emit ExecutorAdded(scheduledExecutionId, executors[i]);
+        }
+
+        // if the scheduler doesn't have a permission to cancel the scheduled action then grant it
+        if (!isCanceler(scheduledExecutionId, sender)) {
+            _addCanceler(scheduledExecutionId, sender);
```

Comment:
I find this snippet a bit confusing. Functionality wise it is correct, but it makes the semantics of calling this function sort of strange (in some cases the account is made a canceler, in some it isn't).

What if instead we didn't have this, and created the canceler at the callsite? I think this will make things extra clear, since in many scenarios we actually won't create a canceler (only in `schedule`, `scheduleGrant` and `scheduleRevoke`). For example:

```
fn setDelay() {
  require(msg.sender == root)
  return _scheduleWithDelay
}

fn schedule() {
  require hasPermission(msg.sender)
  id = _schedule

  // accounts that schedule actions are automatically made cancelers for them, so that they can manage their
  // action. we check that they are not already a canceler since e.g. root may schedule actions (and root is
 // always a global canceler).
  if (!isCanceler(id, msg.sender))
    addCanceler(id, msg.sender)

  return id
}